gcloud-java-logging/src/test/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2ApiTest.java

```diff
@@ -0,0 +1,117 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class LoggingServiceV2ApiTest {
+  private static LocalLoggingHelper loggingHelper;
+  private LoggingServiceV2Api loggingApi;
+
+  @BeforeClass
+  public static void startStaticServer() {
+    loggingHelper = new LocalLoggingHelper("in-process-1");
+    loggingHelper.start();
+  }
+
+  @AfterClass
+  public static void stopServer() {
+    loggingHelper.shutdownNow();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    loggingHelper.reset();
+    ServiceApiSettings settings = new ServiceApiSettings();
+    settings.setChannel(loggingHelper.createChannel());
+    loggingApi = LoggingServiceV2Api.create(settings);
+  }
+
+  @After
+  public void tearDown() {
+    loggingApi.close();
+  }
+
+  @Test
+  public void testWriteLog() {
+    String logName = LoggingServiceV2Api.createLogNamePath("my-project", "my-log");
+    MonitoredResource resource = MonitoredResource.newBuilder().build();
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.newBuilder().setLogName(logName).setTextPayload("foobar").build());
+    loggingApi.writeLogEntries(logName, resource, Collections.<String, String>emptyMap(), entries);
+  }
+
+  @Test
+  public void testListLog() {
+    String logName = LoggingServiceV2Api.createLogNamePath("my-project", "my-log");
+    MonitoredResource resource = MonitoredResource.newBuilder().build();
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.newBuilder().setLogName(logName).setTextPayload("foobar").build());
+    loggingApi.writeLogEntries(logName, resource, Collections.<String, String>emptyMap(), entries);
+
+    Iterable<LogEntry> gotEntries =
+        loggingApi.listLogEntries(Collections.singletonList("my-project"), "", "");
+    List<LogEntry> gotList = new ArrayList<>();
+    for (LogEntry e : gotEntries) {
+      gotList.add(e);
+    }
+    Assert.assertEquals(entries, gotList);
+  }
+
+  @Test
+  public void testListNoLog() {
+    Iterable<LogEntry> gotEntries =
+        loggingApi.listLogEntries(Collections.singletonList("my-project"), "", "");
+    List<LogEntry> gotList = new ArrayList<>();
+    for (LogEntry e : gotEntries) {
+      gotList.add(e);
+    }
+    Assert.assertEquals(Collections.<LogEntry>emptyList(), gotList);
+  }
+
+  @Test
+  public void testDeleteLog() {
+    String logName = LoggingServiceV2Api.createLogNamePath("my-project", "my-log");
+    MonitoredResource resource = MonitoredResource.newBuilder().build();
+    List<LogEntry> entries = new ArrayList<>();
+    entries.add(LogEntry.newBuilder().setLogName(logName).setTextPayload("foobar").build());
+    loggingApi.writeLogEntries(logName, resource, Collections.<String, String>emptyMap(), entries);
+
+    loggingApi.deleteLog(logName);
+
+    Iterable<LogEntry> gotEntries =
+        loggingApi.listLogEntries(Collections.singletonList("my-project"), "", "");
+    List<LogEntry> gotList = new ArrayList<>();
```

Comment:
Nit, alternative:
`assertTrue(Iterables.isEmpty(goEntries))`
or (if, like current code, you want to *show* the extra entries)
`assertTrue(goEntries.toString(), Iterables.isEmpty(goEntries))`

BTW, for converting `Iterable` to `List` you could use `Lists.newArrayList(goEntries)`